### DISCUSSION TITLE
Ultrawide 21:9

### DISCUSSION THREAD

**Discussion by mordes89** [ChristopherKlay/StadiaEnhanced#236] [Ideas & Requests]
First of all, Thank you for creating this extension! You are the best!

This is not an issue, more of a request. Is there a way to have it force an aspect ratio for ultrawide screens (I believe it's 21:9). 
Currently, Stadia does not have an option for that, and I don't see any solutions on various online forums. I did find an extension that forces the ratio, but the width of the screen stays as if it's not stretched. So the sides are cut off.

Any advice you can give me would be welcome. And if you can add this feature to the current extension, I would be eternally thankful!

Cheers!
Mike

**Comment by ChristopherKlay** [comment]
Since Stadia only offers 16:9 output for streams, the only option would be to stretch the image to fit the screen, which would heavily mess with the actual visuals of the game - hence it's not included in the extension yet.

Unless Stadia actually offers support for different resolutions (which is unlikely in the near future), i don't think there is any (good) solution for this.

**Comment by mordes89** [comment]
Thanks for the response! 

I'm not sure it would be so bad to stretch the image. I use an extension that stretches the image on YouTube vids ("Ultrawide Video" extension) and it does not look bad. It is better than the two blank spaces on either side. This is my opinion. I am sure some people would disagree. Do you know a way for me to stretch the image?

**Comment by ChristopherKlay** [comment]
Have you tried using [Ultrawidify](https://chrome.google.com/webstore/detail/ultrawidify/dndehlekllfkaijdlokmmicgnlanfjbi)?

Should give you the options to stretch, without cropping.

**Comment by mordes89** [comment]
I have tried that. I could not get it to work with Stadia. Only thing that
worked was manually changing the Windows resolution while playing, but
sadly the refresh rate on the higher resolution is much lower.

On Thu, Dec 30, 2021 at 4:44 PM Christopher Klay ***@***.***>
wrote:

> Have you tried using Ultrawidify
> <https://chrome.google.com/webstore/detail/ultrawidify/dndehlekllfkaijdlokmmicgnlanfjbi>
> ?
>
> Should give you the options to stretch, without cropping.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ChristopherKlay/StadiaEnhanced/issues/214#issuecomment-1003190665>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APB7CFPD4LOPGRJSDW5P6VTUTTHENANCNFSM5KYRWMKQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Cheers,
Mike

**Comment by ChristopherKlay** [comment]
Just tested a few things.

You can theoretically stretch the stream to any aspect ratio, without cutting off anything, by throwing the following code into the console, while a game is running:
```
document.getElementsByClassName('ma9Vne')[document.getElementsByClassName('ma9Vne').length -1].style.width = '100%'
document.getElementsByClassName('ma9Vne')[document.getElementsByClassName('ma9Vne').length -1].style.height = '100%'
document.getElementsByClassName('vvjGmc')[document.getElementsByClassName('vvjGmc').length -1].style.objectFit = 'fill'
```
But like mentioned above, it will look **heavily** distorted.

**Comment by mordes89** [comment]
Thank you SO much for this! I finally had some time to play around with
this. Sadly, this doesn't let me stretch past 100% which is what I need to
stretch the sides out to the edge of my screen. You can see in the
screenshot, I can make the width and height less than 100%. but I can't
stretch it beyond 100%. Same if I try px instead of a percentage (at 10px,
it's tiny, 1000px and 58000px are the same as 100% and 1500%). I can go
smaller, but at a certain point, it just stops being able to stretch wider.

[image: image.png]

On Fri, Dec 31, 2021 at 11:45 AM Christopher Klay ***@***.***>
wrote:

> Just tested a few things.
>
> You can theoretically stretch the stream to any aspect ratio, without
> cutting off anything, by throwing the following code into the console,
> while a game is running:
>
> document.getElementsByClassName('ma9Vne')[document.getElementsByClassName('ma9Vne').length -1].style.width = '100%'
> document.getElementsByClassName('ma9Vne')[document.getElementsByClassName('ma9Vne').length -1].style.height = '100%'
> document.getElementsByClassName('vvjGmc')[document.getElementsByClassName('vvjGmc').length -1].style.objectFit = 'fill'
>
> But like mentioned above, it will look *heavily* distorted.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ChristopherKlay/StadiaEnhanced/issues/214#issuecomment-1003415663>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APB7CFMBUSBKBEQ2PH5KQGDUTXMZHANCNFSM5KYRWMKQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Cheers,
Mike

**Comment by ChristopherKlay** [comment]
I'll convert this to a feature request / discussion and update it whenever i find a good solution (:
